### PR TITLE
Update the opening of README

### LINKED ISSUE #160: Update README and document front page for a better introduction (why people want this)

### PR DESCRIPTION
**Pull request by xuhdev**
Explain better for why people want PyDAX.

Close #160

---

## Checklist:

<details open>

<summary></summary>

- Does this pull request close an issue? We encourage you to open an issue first if this pull request (PR) is not a
  minor change.
  - [ ] No
    - [ ] The change in this pull request is minor.
  - [x] Yes
    - Close #160

For the following questions, only check the boxes that are applicable.

- [x] Ensure that the pull request text above the horizontal line is descriptive.
- [x] Add/update relevant tests.
- [x] Add/update relevant documents.

- Do you have triage permission of this repository?
  - [ ] No
    - You are good.
  - [x] Yes
    - Does this pull request close an issue?
      - [ ] No, this pull request also acts as an issue by itself.
        - [ ] Assign yourself.
        - [ ] Label this pull request.
        - [ ] Put a milestone.
        - [ ] Put this pull request under "In Progress" or "Under Review" pipeline on ZenHub.
        - [ ] Put this pull request under the appropriate epic on ZenHub.
        - [ ] Put an estimate on ZenHub.
        - [ ] Add dependency relationship with other issues/pull requests on ZenHub.
      - [x] Yes, this pull request addresses an issue and does not act as an issue.
        - [x] Connect this pull request to the underlying issue on ZenHub.
        - [x] DON'T:
          - DON'T assign yourself or anyone else.
          - DON'T label.
          - DON'T put a milestone.
          - DON'T put this pull request under any epic on ZenHub.
          - DON'T put an estimate on ZenHub.
    - [x] When the pull request is ready, request review.
</details>

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -34,11 +34,28 @@ PyDAX (Under Development)
    :target: https://github.com/CODAIT/pydax/commit/master
    :alt: Development Environment
 
-PyDAX is a Python API that enables easy, pragmatic, and elegant programmatical downloading and loading of datasets.
+PyDAX is a Python API that enables data consumers and distributors to easily use and share datasets, and establishes a
+standard for exchanging data assets. It enables:
+
+- a data scientist to have a simpler and more unified way to begin working with a wide range of datasets, and
+- a data distributor to have a consistent, safe, and open source way to share my datasets with interested communities.
```
Comment (edwardleardi) [inline README.rst]:
```suggestion
- a data distributor to have a consistent, safe, and open source way to share datasets with interested communities.
```

Comment (edwardleardi) [reply README.rst]: either change `my` to `their` or remove the word entirely

File: README.rst [hunk 2]
```diff
@@ -34,11 +34,28 @@ PyDAX (Under Development)
...
+
+.. sidebar:: Quick Example
```
Comment (edwardleardi) [inline README.rst:43]: The sidebar formatting looks really weird to me. Is there a way to make this look nicer?

image

Comment (edwardleardi) [reply README.rst:43]: Do we really need this example on top of the Quick Start section? It seems it is repeating content from the Quick Start section, which feels redundant.

File: README.rst [hunk 3]
```diff
@@ -34,11 +34,28 @@ PyDAX (Under Development)
...
+      >>> pydax.list_all_datasets()
+      {'claim_sentences_search': ('1.0.2',),
+       ..., 'wikitext103': ('1.0.1',)}
+      >>> pydax.load_dataset('wikitext103')
+      {...}  # Content of the dataset
 
 Install the Package & its Dependencies
 --------------------------------------
 
+.. TODO: Prior to the first release, replace this section with installing from pypi
```
Comment (edwardleardi) [inline README.rst:57]: we can add this in now can't we?

**Review by edwardleardi — CHANGES_REQUESTED** [review]

File: README.rst (hunk 1 above)
Comment (xuhdev) [reply README.rst]: I removed the word. Saving every single word contributes to succinctness

File: README.rst (hunk 2 above)
Comment (xuhdev) [reply README.rst:43]: I think quickstart section is too long to get someone with technical background to be interested in 2min. The most important lines seem to be buried inside the other features.

The formatting is decided by the theme and the rst compiler and we can't do much about it. We can remove the box entirely though, if that looks better.

File: README.rst (hunk 3 above)
Comment (xuhdev) [reply README.rst:57]: We can do that, but that won't be the latest version. But we can constantly push new versions to pypi if that works for you

File: README.rst (hunk 2 above)
Comment (edwardleardi) [reply README.rst:43]: Is there a setting that controls page width? It seems the problem is the page content area is too narrow squeezing the two code cells into using horizontal scroll bars.

File: README.rst (hunk 3 above)
Comment (edwardleardi) [reply README.rst:57]: ok, probably better to wait until the marketable release then, kind of annoying to have to tag, release, and upload until then.

File: README.rst (hunk 2 above)
Comment (xuhdev) [reply README.rst:43]: I don't see the option in the theme: https://sphinx-rtd-theme.readthedocs.io/en/stable/configuring.html

So I think we have to add a custom CSS

Comment (xuhdev) [reply README.rst:43]: I've added a custom css

Comment (edwardleardi) [reply README.rst:43]: Thanks, that looks better

**Review by edwardleardi — APPROVED** [review]

**Pull request merged**